src/main/java/org/testng/internal/Parameters.java

```diff
@@ -200,6 +257,21 @@ private static void checkParameterTypes(String methodName,
     }
   }
 
+  private static String prettyFormat(List<Class> classes) {
+    StringBuilder builder = new StringBuilder("<");
+    if (classes.size() == 1) {
```

Comment:
The explicit check for `size ==1` was done because if there was just one element in the array then we need a pretty formatted output like `<Foo>` (without the `,`)
Yes the method will fail if the list is empty, but since the list is currently controlled via static data, I thought it would be ok.